### ISSUE TITLE
GrabDragInteraction calling onGrab twice for a single grab

### ISSUE THREAD

**Issue by jbphet** [phetsims/scenery-phet#560]
In https://github.com/phetsims/gravity-force-lab/issues/181 I am working to add sound to the Gravity Force Lab sim.  In the process of trying to add sounds for pickup and release of the ruler, I found that the grab sound was playing twice when the ruler was grabbed using keyboard interaction.  Through the use of debug messages, I could see that `GrabDragInteraction.onGrab` is being called twice with a release in between.  With debug printouts in place, when I navigate using the keyboard to select the ruler, then select it (using the space bar), I see this output all at once in the console:

```
ISLCRulerNode.js?bust=1575583093232:272 onGrab called
ISLCRulerNode.js?bust=1575583093232:170 play called on grabRulerSoundPlayer
ISLCRulerNode.js?bust=1575583093232:279 onRelease called
ISLCRulerNode.js?bust=1575583093232:272 onGrab called
ISLCRulerNode.js?bust=1575583093232:170 play called on grabRulerSoundPlayer
ISLCRulerNode.js?bust=1575583093232:279 onRelease called
```

I showed this to @zepumph over Slack and he said he'd investigate, so I'm assigning to him.

**Comment by terracoda** [comment]
Adding screen shot that on grab I get both a release alert and a grabbed alert
![Screen Shot 2019-12-09 at 4 05 03 PM](https://user-images.githubusercontent.com/1902116/70466510-dee5a100-1a9d-11ea-9a4a-ab88a929510b.png)

**Comment by jessegreenberg** [comment]
Since this is related to GrabDragInteraction I can also take a look at this.

**Comment by jessegreenberg** [comment]
This is caused by the addition of PressListener to GrabDragInteraction. PressListener adds pointer compatibility to GrabDragInteraction. But PressListener `release` fires for both pointer and keyboard events.

**Comment by jessegreenberg** [comment]
Changing the PressListener so that it only responds to Pointer events fixes the issue.
```js
      // on pointer down, switch to "draggable" representation in the PDOM - necessary for accessible tech that
      // uses pointer events like iOS VoiceOver
      const pressListener = new PressListener( {
        press: ( event, listener ) => {
          if ( !event.isA11y() ) {
            this.releaseDraggable();
            this.onGrab();
          }
        },
        release: ( event, listener ) => {
          if ( !event.isA11y() ) {
            this.releaseDraggable();
          }
        },
```

But I wonder if we should change GrabDragInteraction to leverage PressListener for the key listeners too.

**Comment by jessegreenberg** [comment]
I tried leveraging PressListener for more of the PDOM related input too. We cannot use it for everything because PressListener uses `click` events for PDOM input and we need to listen to `keydown`/`keyup` input when the target node is a `div`. But we can move the portion that listens to `click` to PressListener, this change is working pretty well:
```diff
diff --git a/js/accessibility/GrabDragInteraction.js b/js/accessibility/GrabDragInteraction.js
index 180dc491..8a9dedc9 100644
--- a/js/accessibility/GrabDragInteraction.js
+++ b/js/accessibility/GrabDragInteraction.js
@@ -292,43 +292,9 @@ define( require => {
       // the draggable, so guard against that.
       let guardKeyPressFromDraggable = false;
 
-      // when the "Grab {{thing}}" button is pressed, focus the draggable node and set to dragged state
+      // when the "Grab {{thing}}" button is blurred, make the grab cue invisible if we have had a successful
+      // interaction enough times
       const grabButtonListener = {
-        click: () => {
-
-          // don't turn to draggable on mobile a11y, it is the wrong gesture - user should press down and hold
-          // to initiate a drag
-          if ( supportsGestureA11y() ) {
-            return;
-          }
-
-          // if the draggable was just released, don't pick it up again until the next click event so we don't "loop"
-          // and pick it up immediately again.
-          if ( !guardKeyPressFromDraggable ) {
-            this.turnToDraggable();
-
-            this.node.focus();
-
-            this.onGrab();
-
-            // Add the newly created focusHighlight to the scene graph if focusHighlightLayerable, just like the
-            // original focus highlight was added. By doing this on click, we make sure that the node's
-            // focusHighlight has been completely constructed (added to the scene graph) and can use its parent. But only
-            // do it once.
-            if ( node.focusHighlightLayerable ) {
-              assert && assert( this.grabFocusHighlight.parent, 'how can we have focusHighlightLayerable with a ' +
-                                                                'node that is not in the scene graph?' );
-              // If not yet added, do so now.
-              if ( !this.grabFocusHighlight.parent.hasChild( this.dragFocusHighlight ) ) {
-                this.grabFocusHighlight.parent.addChild( this.dragFocusHighlight );
-              }
-            }
-          }
-
-          // "grab" the draggable on the next click event
-          guardKeyPressFromDraggable = false;
-        },
-
         blur: () => {
           if ( this.numberOfGrabs >= options.grabsToCue ) {
             this.grabCueNode.visible = false;
@@ -382,11 +348,54 @@ define( require => {
       // on pointer down, switch to "draggable" representation in the PDOM - necessary for accessible tech that
       // uses pointer events like iOS VoiceOver
       const pressListener = new PressListener( {
-        press: () => {
-          this.turnToDraggable();
-          this.onGrab();
+        press: ( event, listener ) => {
+
+          // If we are responding to a press from an PDOM event we are in our "grabbable" state receiving a `click`
+          // event, so turn to draggable
+          if ( event.isA11y() ) {
+
+            // don't turn to draggable on mobile a11y, it is the wrong gesture - user should press down and hold
+            // to initiate a drag and we will then go through non-a11y Pointer events
+            if ( supportsGestureA11y() ) {
+              return;
+            }
+
+            // if the draggable was just released, don't pick it up again until the next click event so we don't "loop"
+            // and pick it up immediately again.
+            if ( !guardKeyPressFromDraggable ) {
+              this.turnToDraggable();
+
+              this.node.focus();
+
+              this.onGrab();
+
+              // Add the newly created focusHighlight to the scene graph if focusHighlightLayerable, just like the
+              // original focus highlight was added. By doing this on click, we make sure that the node's
+              // focusHighlight has been completely constructed (added to the scene graph) and can use its parent. But only
+              // do it once.
+              if ( node.focusHighlightLayerable ) {
+                assert && assert( this.grabFocusHighlight.parent, 'how can we have focusHighlightLayerable with a ' +
+                                                                  'node that is not in the scene graph?' );
+                // If not yet added, do so now.
+                if ( !this.grabFocusHighlight.parent.hasChild( this.dragFocusHighlight ) ) {
+                  this.grabFocusHighlight.parent.addChild( this.dragFocusHighlight );
+                }
+              }
+            }
+
+            // "grab" the draggable on the next click event
+            guardKeyPressFromDraggable = false;
+          }
+          else {
+            this.releaseDraggable();
+            this.onGrab();
+          }
+        },
+        release: ( event, listener ) => {
+          if ( !event.isA11y() ) {
+            this.releaseDraggable();
+          }
         },
-        release: () => this.releaseDraggable(),
 
         // this listener shouldn't prevent the behavior of other listeners, and this listener should always fire
         // whether or not the pointer is already attached
```

But I don't think it is any better than https://github.com/phetsims/scenery-phet/issues/560#issuecomment-564307914. So I am going to proceed with that that and clarify that the PressListener here should only respond to non-PDOM events.

**Comment by jessegreenberg** [comment]
Commit made above. Only difference from https://github.com/phetsims/scenery-phet/issues/560#issuecomment-564307914 is documentation and we also `releaseDraggable` if event is `null` because this indicates that PressListener was interrupted.

@zepumph can you please review this since you are familiar with GrabDragInteraction? Take a look at the above comment and let me know if you think I made the wrong call.

**Comment by zepumph** [comment]
Yeah that makes sense. Good work!
